### ISSUE TITLE
Question: What is the expected behavior on rejected promises?

### ISSUE THREAD

**Issue by mictian** [janryWang/callbag-map-promise#2]
Hi,

First of all, beautiful, simple and very useful package, thank you :smile: !
I wounder what should be the expected behavior for rejected promises?

Given the following code

```js
const { pipe } = require('callbag-basics');
const subscribe = require('callbag-subscribe');
const mapPromise = require('callbag-map-promise').default;
const tap = require('callbag-tap');

pipe(
  starter, // anything like from-event
  mapPromise((v) => {
    return Promise.reject(new Error(v));
  }),
  tap(v => console.log('Value returned from promise', v)),
  subscribe({
    next: v => console.log('NEXT', v),
    error: e => console.log('ERROR', e),
    complete: () => console.log('COMPLETE')
  })
);
```

the flow get stuck after `mapPromise`. Am I using it in the wrong way?

My poor understanding of there flow tells me that the `flat-map` [here](https://github.com/avinashcodes/callbag-flat-map/blob/master/index.js#L35) is no correctly handling the rejection from `from-promise` [here](https://github.com/staltz/callbag-from-promise/blob/master/index.js#L12).
But in any case, is this package who use them.

Long story short, how should rejected promises be handled?

Regards,
Mictian
